Sources/ApolloCodegenLib/Glob.swift

```diff
@@ -185,7 +186,15 @@ public struct Glob {
     }
 
     return OrderedSet<String>(directories.compactMap({ directory in
-      var path = directory.appendingPathComponent(lastPart).standardizedFileURL.path
+      // build directory URL up the the '*' in the pattern, and check if it is a symlink, if so skip it
+      let symCheckURL = directory.appendingPathComponent(lastDirComponent).standardizedFileURL
```

Comment:
@AnthonyMDev this block can run the same symlink check for production use cases we previously had in the `ApolloCodegen` class. But keeps all symlink related checks contained within this `compactMap` call

Reply:
Alternatively, I may be able to build the url up to the `*` in the pattern like we are doing here, and then run the `.resolvingSymlinksInPath` on that URL, and then appending the final path component which could cover both production and test use cases without needing to access resource keys and result in less code. Going to look into that now.

Reply:
Would look something like this although maybe cleaned up a little:
```
let last = parts.joined(separator: String.Globstar)
let lastParts = last.components(separatedBy: "*")
let lastDirComponent = lastParts.first ?? ""
let lastPart = lastParts.last ?? ""
....
....
....
let url = directory.appendingPathComponent(lastDirComponent).standardizedFileURL.resolvingSymlinksInPath()
var path = url.appendingPathComponent("*\(lastPart)").standardizedFileURL.path
```